File: src/SagaGuide/SagaGuide.Core/Domain/Features/IFeature.cs
namespace SagaGuide.Core.Domain.Features;

public interface IFeature
{
    public enum FeatureTypeEnum
    {
        AttributeBonus,
        ConditionalModifierBonus,
        DamageReductionBonus,
        ReactionBonus,
        SkillBonus,
        SkillPointBonus,
        SpellBonus,
        SpellPointBonus,
        WeaponBonus,
        WeaponDrDivisorBonus,
        AttributeCostReduction,
        ContainedWeightReduction
    }
    
    FeatureTypeEnum FeatureType { get; }
}

File: src/frontend/src/domain/interfaces/feature/IFeature.ts
export interface IFeature
{
    featureType: FeatureTypeEnum,
    amount: number,
    isScalingWithLevel: boolean,
}

export enum FeatureTypeEnum
{
    AttributeBonus = "AttributeBonus",
    ConditionalModifierBonus = "ConditionalModifierBonus",
    DamageReductionBonus = "DamageReductionBonus",
    ReactionBonus = "ReactionBonus",
    SkillBonus = "SkillBonus",
    SkillPointBonus = "SkillPointBonus",
    SpellBonus = "SpellBonus",
    SpellPointBonus = "SpellPointBonus",
    WeaponBonus = "WeaponBonus",
    WeaponDrDivisorBonus = "WeaponDrDivisorBonus",
    AttributeCostReduction = "AttributeCostReduction",
    ContainedWeightReduction = "ContainedWeightReduction",
}

